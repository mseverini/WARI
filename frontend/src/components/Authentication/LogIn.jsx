import React from 'react'
import {graphql} from 'react-apollo'
import LogInForm from './LogInForm'
import gql from 'graphql-tag'

const logInMutation = gql`
mutation LogInUser($credentials: AuthProviderCredentials) {
  loginUser(credentials: $credentials)
  {token}
}`

class LogIn extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: '',
      mutate: props.mutate,
      loggedIn: false,
    }

    this.handleChange = this.handleChange.bind(this)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleChange(e) {
    const target = e.target
    const value = target.type === 'checkbox' ? target.checked : target.value
    const name = target.name

    this.setState({
      [name]: value
    })
  }


  async handleSubmit(e) {
    e.preventDefault()

    const credentials = {
      email: this.state.email,
      password: this.state.password,
    }

    let response = await this.state.mutate({
      variables: {
        credentials,
      },
    })

    let token = response.data.loginUser.token
    global.sessionStorage.setItem('token', token)
    window.location.reload()
  }


  render() {
    return (<LogInForm handleChange={this.handleChange} handleSubmit={this.handleSubmit}/>)
  }
}

const LogInWithMutation = graphql(logInMutation)(LogIn)
export default LogInWithMutation
