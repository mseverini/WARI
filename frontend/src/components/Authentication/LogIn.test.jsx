import React from 'react'
import {shallow} from 'enzyme'
import LogIn from './LogIn'

test("Renders the Login component", ()=>{
  let actual = shallow(<LogIn/>)
  expect(actual).toMatchSnapshot()
})

