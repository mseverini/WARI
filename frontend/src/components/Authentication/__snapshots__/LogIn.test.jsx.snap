// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Login component 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <LogIn
    mutate={[Function]}
/>,
  "nodes": Array [
    <LogIn
      mutate={[Function]}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_context": Object {},
    "_element": <Apollo(LogIn) />,
    "_instance": GraphQL {
      "context": Object {},
      "dataForChildViaMutation": [Function],
      "previousData": Object {},
      "props": Object {},
      "refs": Object {},
      "renderedElement": <LogIn
        mutate={[Function]}
/>,
      "setWrappedInstance": [Function],
      "shouldRerender": false,
      "state": Object {},
      "type": 1,
      "updater": Updater {
        "_renderer": [Circular],
      },
      "version": 0,
    },
    "_newState": null,
    "_rendered": <LogIn
      mutate={[Function]}
/>,
    "_rendering": false,
    "_updater": Updater {
      "_renderer": [Circular],
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Apollo(LogIn) />,
}
`;
